amp/WEB-INF/src/org/digijava/module/aim/dbentity/AmpActivityFields.java

```diff
@@ -2172,5 +2175,17 @@ public void addCostAmount(AmpFundingAmount costAmount) {
             this.costAmounts.add(costAmount);
         }
 
+    @Override
+    public void touch() {
+        Date updateDate = Calendar.getInstance().getTime();
+        setUpdatedDate(updateDate);
+        setModifiedDate(updateDate);
+
+        AmpTeamMember currentAmpTeamMember = TeamMemberUtil.getCurrentAmpTeamMember(TLSUtils.getRequest());
```

Comment:
There is an assumption that teamMember passed to org.dgfoundation.amp.onepager.util.ActivityUtil#saveActivityNewVersion coincides with the one from current http request. Also activities could be saved outside of http request scope.

Reply:
For example in AMP Offline one user can save an activity on behalf of other user.
Also, we might need the possibility to save many activities on behalf on many different users inside same http request and/or tranzaction.